emcc.py

```diff
@@ -1143,8 +1143,11 @@ def check(input_file):
           # if not wasm-only, we can't do async compilation as the build can run in other
           # modes than wasm (like asm.js) which may not support an async step
           shared.Settings.BINARYEN_ASYNC_COMPILATION = 0
+          warning = 'This will reduce performance and compatibility (some browsers limit synchronous compilation), see https://github.com/kripken/emscripten/wiki/WebAssembly#codegen-effects'
           if 'BINARYEN_ASYNC_COMPILATION=1' in settings_changes:
-            logging.warning('BINARYEN_ASYNC_COMPILATION requested, but disabled since not in wasm-only mode')
+            logging.warning('BINARYEN_ASYNC_COMPILATION requested, but disabled because of user options. ' + warning)
+          elif 'BINARYEN_ASYNC_COMPILATION=0' not in settings_changes:
+            logging.warning('BINARYEN_ASYNC_COMPILATION disabled due to user options. ' + warning)
```

Comment:
Hm, ok. but it would still be nice if we could come up with a warning a little more actionable than "play around with random settings and see if the warning goes away". I don't really have any ideas right now though.